src/smc.jl

```diff
@@ -38,10 +38,10 @@ function AbstractMCMC.sample(
 )
     if !isempty(kwargs)
         @warn "keyword arguments $(keys(kwargs)) are not supported by `SMC`"
-    end
+			end
 
     # Create a set of particles.
-    particles = ParticleContainer([Trace(model) for _ in 1:sampler.nparticles])
+    particles = ParticleContainer([Trace(model, rng) for _ in 1:sampler.nparticles])
```

Comment:
I think it is crucial to not use the user-provided RNG directly if one wants to keep track of the number of generated random numbers in order to achieve reproducibility: if the user updates the state of the RNG outside of the trace (e.g., by generating other random numbers, in particular if it is just the default RNG) the sequence of random numbers in the trace can not be reproduced anymore.

Reply:
Good point, maybe we can make a copy of the rng and pass it to Trace ?